roles/gpu/tasks/gpu.yml

```diff
@@ -0,0 +1,99 @@
+---
+- name: Install yum requirements for gpu driver installation
+  ansible.builtin.yum:
+    state: 'installed'
+    update_cache: true
+    name:
+      - tar
+      - bzip2
+      - make
+      - automake
+      - gcc
+      - gcc-c++
+      - pciutils
+      - elfutils-libelf-devel
+      - libglvnd-devel
+      - bind-utils
+      - wget
+  become: true
+
+- name: Gather facts to get the latest kernel version
+  ansible.builtin.setup:
+  become: true
+
+- name: Install kernel developement package matching running kernel version
+  ansible.builtin.yum:
+    name: 'kernel-devel-{{ ansible_kernel }}'
+  register: yum_result
+  failed_when: yum_result.rc != 0
+  when: true
+  become: true
+
+- name: Download a driver installation file from NVidia
+  ansible.builtin.get_url:
+    url: '{{ gpu_url_dir }}/{{ gpu_runfile }}'
+    dest: '/root/{{ gpu_runfile }}'
+    mode: '0700'
+  become: true
+
```

Comment:
```suggestion
  when: (smi_gpu_count.stdout | default([]) | lower | regex_findall('nvidia') | length != gpu_count) or
             (smi_driver_version.stdout  | default([]) != gpu_driver_version)

```

Reply:
again, this is not needed if we avoid move into `gpu.yml`